libqtile/backend/__init__.py

```diff
@@ -1,14 +1,45 @@
+from __future__ import annotations
+
 import importlib
+import importlib.util
+from typing import TYPE_CHECKING
 
 from libqtile.utils import QtileError
 
-CORES = [
-    "wayland",
-    "x11",
-]
+if TYPE_CHECKING:
+    from typing import Any
+
+    from libqtile.backend.base import Core
+
+
+CORES = {
+    "wayland": ("wlroots",),
```

Comment:
Why not listing pywayland and pywlroots here?

Reply:
`wlroots` is the name of the `pywlroots` module (and we test by importing), and pywlroots pulls in `pywayland` anyway